sql/core/src/main/scala/org/apache/spark/sql/execution/SparkSqlParser.scala

```diff
@@ -122,9 +122,10 @@ class SparkSqlAstBuilder extends AstBuilder {
     // Create the explain comment.
     val statement = plan(ctx.statement)
     if (isExplainableStatement(statement)) {
-      ExplainCommand(statement, extended = options.exists(_.EXTENDED != null))
+      ExplainCommand(statement, extended = options.exists(_.EXTENDED != null),
+        codegen = options.exists(_.CODEGEN != null))
     } else {
-      ExplainCommand(OneRowRelation)
+      ExplainCommand(OneRowRelation, codegen = options.exists(_.CODEGEN != null))
```

Comment:
is there a point showing the generated code of an otherwise unexplainable command?